### ISSUE TITLE
NLTK downloader bug

### ISSUE THREAD

**Issue by stevenbird** [nltk/nltk#882]
```
$ sudo python -m nltk.downloader
Traceback (most recent call last):
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/runpy.py", line 170, in _run_module_as_main
    "__main__", mod_spec)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Users/sb/git/nltk/nltk/downloader.py", line 2266, in <module>
    halt_on_error=options.halt_on_error)
  File "/Users/sb/git/nltk/nltk/downloader.py", line 655, in download
    self._interactive_download()
  File "/Users/sb/git/nltk/nltk/downloader.py", line 967, in _interactive_download
    DownloaderGUI(self).mainloop()
  File "/Users/sb/git/nltk/nltk/downloader.py", line 1227, in __init__
    self._fill_table()
  File "/Users/sb/git/nltk/nltk/downloader.py", line 1523, in _fill_table
    items = self._ds.collections()
  File "/Users/sb/git/nltk/nltk/downloader.py", line 499, in collections
    self._update_index()
  File "/Users/sb/git/nltk/nltk/downloader.py", line 854, in _update_index
    packages[child.id] = child
AttributeError: 'str' object has no attribute 'id'
```

**Comment by kaselis** [comment]
The same here, started this morning only. nltk version 2.0.4

**Comment by boffbowsh** [comment]
Ditto, with 3.0.0 and 3.0.1 here

**Comment by alexlouden** [comment]
+1 same here

Looks like collections contains `lang_id` which doesn't have a package definition:

```
  <collections>
    <collection id="all-corpora" name="All the corpora">
      ....
      <item ref="langid" />
```

Here's the culprit: https://github.com/nltk/nltk_data/blob/fa7aae1c5c5e26b3cc3217dae15005c471184819/collections/all-corpora.xml#L29

I'll make a pull request

**Comment by stevenbird** [comment]
I dealt with [the spurious langid line](https://github.com/nltk/nltk_data/blob/fa7aae1c5c5e26b3cc3217dae15005c471184819/collections/all-corpora.xml#L29) in [this commit](https://github.com/nltk/nltk_data/commit/dba94eb6d16f8367f1b3631eeee73a4bffb94ba2) but the downloader is still broken.

**Comment by alexlouden** [comment]
I think this is the problem:
`<item ref="bllip_wsj_no_aux.xml"/>`

Could you remove the `.xml`?

https://github.com/nltk/nltk_data/blob/23a9666955ec0c9bedeb25dc4903bcd8d9caa8ae/collections/all.xml#L88

**Comment by stevenbird** [comment]
Oops. Removed in https://github.com/nltk/nltk_data/commit/1528e595716bc9001a7a7f3bbdea55b3406c9842

**Comment by stevenbird** [comment]
OK, all working again, thanks @alexlouden!

**Comment by alexlouden** [comment]
No worries - thanks for your fast fixes!

**Comment by zachguo** [comment]
I ran into this bug this morning, using `nltk` that came with `anaconda`.

```
removing collection member with no package: customer_reviews_1
Traceback (most recent call last):
  File "/opt/conda/lib/python2.7/runpy.py", line 162, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/opt/conda/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/opt/conda/lib/python2.7/site-packages/nltk/downloader.py", line 2267, in <module>
    halt_on_error=options.halt_on_error)
  File "/opt/conda/lib/python2.7/site-packages/nltk/downloader.py", line 664, in download
    for msg in self.incr_download(info_or_id, download_dir, force):
  File "/opt/conda/lib/python2.7/site-packages/nltk/downloader.py", line 534, in incr_download
    try: info = self._info_or_id(info_or_id)
  File "/opt/conda/lib/python2.7/site-packages/nltk/downloader.py", line 508, in _info_or_id
    return self.info(info_or_id)
  File "/opt/conda/lib/python2.7/site-packages/nltk/downloader.py", line 875, in info
    self._update_index()
  File "/opt/conda/lib/python2.7/site-packages/nltk/downloader.py", line 857, in _update_index
    packages[child.id] = child
AttributeError: 'str' object has no attribute 'id'
```

**Comment by ahjones** [comment]
I've just had the same issue as @zachguo . I'm new to Python so I'm only able to hack a fix, but here's how I did it. Hopefully this'll be useful either for a quick workaround for others in the same position, or for finding the root cause. I've made the below changes to tag 3.0.4.

I added `print(type(child))` to line 857 of `downloader.py`. The error occurs when the type of `child` is a string rather than a `Package`.

I've edited line 855 so that rather than extending queue to all children of child, it's only extended to children that are of type `Package`. This is the new line: `queue.extend([x for x in child.children if type(x) is Package])`.

I think this isn't a proper solution, but it got me up and running.
